engine/src/main/java/org/terasology/engine/rendering/nui/skin/UISkinFormat.java

```diff
@@ -145,7 +145,8 @@ public void apply(UISkinBuilder builder) {
                         builder.setElementClass(metadata.getType());
                         entry.getValue().apply(builder);
                     } else {
-                        logger.warn("Failed to resolve UIWidget class {}, skipping style information", entry.getKey());
+                        logger.atWarn().addArgument(() -> entry.getKey()).
+                                log("Failed to resolve UIWidget class {}, skipping style information");
```

Comment:
`entry.getKey()` is calculated already before the logger call.
Please create and reference a local variable.